### ISSUE TITLE
Ceate a demo for Muse in PA

### ISSUE THREAD

**Issue by casaout** [sealuzh/PersonalAnalytics#169]
- [ ] check if the UWP Windows Library can be used within a WPF application (using a portable library wrapper): http://developer.choosemuse.com/developer-sdk-windows/getting-started-with-libmuse-windows 
- [ ] create demo (e.g. 10 blinks color change)

**Comment by s-mueller** [comment]
Should we create the demo in a new PA branch or a standalone project?

**Comment by casaout** [comment]
I think on a separate branch on PA. Creating it on a new PA branch would make it easier to re-use most of what we already have & save time. We just have to make sure to NOT merge it to any other branch ever (as long as the demo stuff, e.g. 10 eye blinks are in there). My suggestion is to copy over the MuseTracker from the other branch (to the private repo) and save the current flow-state to the database table (similar to FlowTracker) - and then the light will change the color automatically, without any modifications (keep in mind there is a smoothing function that might need to be disabled for the demo).

Thoughts?

**Comment by s-mueller** [comment]
Ok, the I'll first check whether the SDK can be used in a portable library and if so, I'll follow your advice. One more question though: Won't that include a delay between writing the data to the DB and updating the light?

**Comment by casaout** [comment]
yes, there is a delay (I think up to 3 mins - @manulita knows the details). But we could add an exception to the Flowlight for the demo that we can enable with a parameter to circumvent the smoothing function.

**Comment by s-mueller** [comment]
There are two good resources that explain how to use the Muse Windows SDK. Here is an explanation on how to get the examples running:

http://developer.choosemuse.com/developer-sdk-windows/getting-started-with-libmuse-windows

And here an explanation on how to create your own project:

http://developer.choosemuse.com/developer-sdk-windows/making-your-own-application

I tried the SDK and it works pretty well. It also has blink detection included and in my tests it was very accurate.

There is only one problem. To use it in PA, we have to somehow make it run in WPF projects on Windows 7. So far I did not find a way how to do that, since the library is written in C++ and targeted for UWP apps. I tried the same approach we used for the BLE problem, i.e. creating a portable library. However, I couldn't find a way how to create a portable C++ library instead of a C# one.
